prov/usnic/src/usdf_fabric.c

```diff
@@ -75,9 +76,56 @@
 
 struct usdf_usnic_info *__usdf_devinfo;
 
+static char *
+get_fabric_name(uint32_t version, struct usd_device_attrs *dap)
+{
+	char *bufp;
+	struct in_addr in;
+	char *addrnetw;
+
+	if (FI_VERSION_GE(version, FI_VERSION(1, 4))) {
+		in.s_addr = dap->uda_ipaddr_be & dap->uda_netmask_be;
+		addrnetw = inet_ntoa(in);
+		asprintf(&bufp, "%s/%d", addrnetw, dap->uda_prefixlen);
+	} else {
+		bufp = strdup(dap->uda_devname);
+	}
+
+	return bufp;
+}
+
+static bool
+check_fabric_name(uint32_t version, struct usd_device_attrs *dap, char *hint) {
+	bool valid = false;
+	char *reference;
+
+	USDF_DBG("checking devname: version=%d, devname='%s'\n", version, hint);
+
+	if (version) {
+		reference = get_fabric_name(version, dap);
+		if (strcmp(reference, hint) == 0)
+			valid = true;
+		else
+			USDF_DBG("hint %s failed to match %s\n", hint, reference);
+
+		free(reference);
+		return valid;
+	}
+
+	/* The hint string itself is kind of a version check, in pre-1.4 the
+	* name was just 'usnic'. In 1.4 and beyond, then name is actually CIDR
```

Comment:
Also, my fault this comment is wrong. In pre-1.4 the name wasn't just 'usnic' it was the device name.